File: frontend/src/api/studySession.ts
import request from './request'

export interface StudySession {
  id: number
  knowledge_base_id: number | null
  knowledge_base_name?: string | null
  knowledge_point_id: number | null
  knowledge_point_title?: string | null
  started_at: string
  ended_at: string | null
  total_count: number
  correct_count: number
  accuracy_rate: number
}

export function createStudySession(data: {
  knowledge_base_id?: number
  knowledge_point_id?: number
  total_count?: number
}) {
  return request.post('/api/study-sessions', data) as Promise<StudySession>
}

export function finishStudySession(id: number, data: { total_count: number; correct_count: number }) {
  return request.post(`/api/study-sessions/${id}/finish`, data) as Promise<StudySession>
}

File: backend/app/schemas/study_session.py
from datetime import datetime
from pydantic import BaseModel, Field


class StudySessionCreate(BaseModel):
    knowledge_base_id: int | None = None
    knowledge_point_id: int | None = None
    total_count: int = Field(default=0, ge=0)


class StudySessionFinish(BaseModel):
    total_count: int = Field(ge=0)
    correct_count: int = Field(ge=0)


class StudySessionResponse(BaseModel):
    id: int
    knowledge_base_id: int | None
    knowledge_point_id: int | None
    started_at: datetime
    ended_at: datetime | None
    total_count: int
    correct_count: int
    accuracy_rate: float

    class Config:
        from_attributes = True
